tools/hlslpm/issue.py

```diff
@@ -0,0 +1,347 @@
+from dataclasses import dataclass, field
+from datetime import datetime, date
+from enum import Enum
+import re
+from typing import Dict, List, Optional, Union
+import json
+
+class Category(Enum):
+    NoCategory = None
+    Item = "Item"
+    Deliverable = "Deliverable"
+    Workstream = "Workstream"
+    WorkstreamMilestone = "Workstream Milestone"
+    ProjectMilestone = "Project Milestone"
+
+
+class IssueSection:
+    title: str = None
+    contents: List[str]
+
+    def __init__(self):
+        self.contents = []    
+
+    def getReferenceFromTitle(self):
+        if not self.title:
+            return None
+
+        m = re.match(".*\((.+)\)", self.title)
```

Comment:
I think [named groups](https://docs.python.org/3/library/re.html#re.Match.group) make the regex more readable.

Reply:
I agree in some cases, but I don't think any of these regexes quite get there - named groups also clutter the regex quite a bit.